src/main/scala/scorex/core/network/NodeViewSynchronizer.scala

```diff
@@ -39,22 +40,24 @@ class NodeViewSynchronizer[P <: Proposition, TX <: Transaction[P], SI <: SyncInf
  syncInfoSpec: SIS,
  networkSettings: NetworkSettings) extends Actor with ScorexLogging {
 
-  import History.HistoryComparisonResult._
   import NodeViewSynchronizer._
+  import History.HistoryComparisonResult._
 
   type Timestamp = Long
 
   protected val deliveryTimeout: FiniteDuration = networkSettings.deliveryTimeout
   protected val maxDeliveryChecks: Int = networkSettings.maxDeliveryChecks
   protected val deliveryTracker = new DeliveryTracker(context, deliveryTimeout, maxDeliveryChecks, self)
 
-  /*
+  /**
     We cache peers along with their statuses (whether another peer is ahead or behind of ours,
     or comparison is not possible, or status is not yet known)
-   */
+   **/
   protected val statuses = mutable.Map[ConnectedPeer, HistoryComparisonResult.Value]()
   protected val statusUpdated = mutable.Map[ConnectedPeer, Timestamp]()
-  private var lastSyncInfoSentTime = NetworkTime.time()
+
+  protected var stableSyncRegime = false
+  protected var scheduler: Cancellable = _
```

Comment:
I think this is bug-prone, because different parts of the code may re-assign `scheduler` and, therefore, other parts of the code may end up canceling the wrong scheduler. If possible, I think it would be worth refactoring this in a way that protects ourselves from future bugs of this kind.

Reply:
For now I see that the only time the scheduler is re-assigned is immediately after the scheduler is canceled, in lines 195--196. So, there is no bug of the kind described above. But my concern is about potential future bugs.